### PR TITLE
K8SPXC-233 document the object surgery crash recovery

### PR DESCRIPTION
**Pull request by fiowro**

### CODE REVIEW

File: source/recovery.rst [hunk 1]
```diff
@@ -42,3 +42,121 @@ Applying this option forces the cluster to start. However, there may exist data
...
+      seqno:   19
+      safe_to_bootstrap: 0
+
+   Now find the Pod with the largest ``seqno`` (it is ``cluster1-pxc-2`` in the
+   above example).
+
+5. Now execute the following commands *in a separate shell* to start this node:
+
+   .. code-block:: bash
+
```
Comment (delgod) [inline source/recovery.rst]: let's add recovery
```suggestion

      $ kubectl exec cluster1-pxc-2 -- mysqld --wsrep_recover
```

File: source/recovery.rst [hunk 2]
```diff
@@ -42,3 +42,122 @@ Applying this option forces the cluster to start. However, there may exist data
...
+         $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc5.7-debug"}}}'
+
+2.  Restart all Pods:
+
+   .. code-block:: bash
+
+      $ $ for i in $(seq 0 $(($(kubectl get pxc cluster1 -o jsonpath='{.spec.pxc.size}')-1))); do kubectl delete pod cluster1-pxc-$i --force --grace-period=0; done
+
+3. Wait until the Pod ``0`` is ready, and execute the following code (it is
```
Comment (fiowro) [inline source/recovery.rst]: In the updated doc we have restarted not *Pod 0*, but *all Pods* on the previous step. Now we wait until "Pod 0" is ready and execute some code (perhaps for all pods). Shouldn't we write here "Wait until all Pods are ready"?

Comment (delgod) [reply source/recovery.rst]: no, it is ok

File: source/recovery.rst [hunk 3]
```diff
@@ -42,3 +42,122 @@ Applying this option forces the cluster to start. However, there may exist data
...
+which is based on three PXC Pods.
+
+.. note:: The following commands are written for PXC 8.0. The same steps are
+   also for PXC 5.7 unless specifically indicated otherwise.
+
+1. Change the normal PXC image inside the cluster object to the debug image:
+
+   .. code-block:: bash
+
+      $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc8.0-debug"}}}'
```
Comment (delgod) [inline source/recovery.rst]:
```suggestion
      $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0-debug"}}}'
```

File: source/recovery.rst [hunk 4]
```diff
@@ -42,3 +42,122 @@ Applying this option forces the cluster to start. However, there may exist data
...
+
+   .. note:: For PXC 5.7 this command should be as follows:
+
+      .. code-block:: bash
+
+         $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc5.7-debug"}}}'
```
Comment (delgod) [inline source/recovery.rst]:
```suggestion
         $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"percona/percona-xtradb-cluster-operator:1.4.0-pxc5.7-debug"}}}'
```

File: source/recovery.rst [hunk 5]
```diff
@@ -42,3 +42,122 @@ Applying this option forces the cluster to start. However, there may exist data
...
+
+   The ``mysqld`` process will initialize the database once again, and it will
+   be available for the incoming connections.
+
+6. Now go back to the original PXC image because the debug image is no longer
+   needed:
+
+   .. code-block:: bash
+
+      $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc8.0"}}}'
```
Comment (delgod) [inline source/recovery.rst]:
```suggestion
      $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0"}}}'
```

File: source/recovery.rst [hunk 6]
```diff
@@ -42,3 +42,122 @@ Applying this option forces the cluster to start. However, there may exist data
...
+
+   .. note:: For PXC 5.7 this command should be as follows:
+
+      .. code-block:: bash
+
+         $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc5.7"}}}'
```
Comment (delgod) [inline source/recovery.rst]:
```suggestion
         $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"percona/percona-xtradb-cluster-operator:1.4.0-pxc5.7"}}}'
```

File: source/recovery.rst [hunk 7]
```diff
@@ -42,3 +42,122 @@ Applying this option forces the cluster to start. However, there may exist data
...
+
+7. Go back *to the previous shell* and restart all Pods besides the
```
Comment (delgod) [inline source/recovery.rst]: I think we need to switch to prev shell in step 6.
because shell in step 5 become locked by mysqld

**Pull request merged**